src/lib/ipfs.ts

```diff
@@ -0,0 +1,104 @@
+import promisify = require("es6-promisify");
+import axios from "axios";
+import * as IPFS from 'ipfs-mini';
+const bs58 = require('bs58');
+
+// github.com/github/fetch/issues/175
+function timeout(promise, ms) {
+  return new Promise(function(resolve, reject) {
+    setTimeout(function() {
+      reject(new Error("timeout"))
+    }, ms)
+    promise.then(resolve, reject)
+  })
+}
+
+const ipfsHashPrefix = '1220';
+// Stores a static snapshot of the proposal data from before the ipfs migration
+const legacyIpfsHash = 'QmNaTgnLUAWneQwifeeLZHc5N1k2wqFKSfWPRjTS9ZjTxZ';
+
+let host;
+let protocol;
+try {
+  host = window.location.host;
+  protocol = 'https'
+} catch (e) {
+  // the cacher accesses this docker image directly
+  host = process.env.IPFS_URL;
+  protocol = 'http'
+}
+
+const ipfsApi: any = new IPFS({
+  protocol,
+  host,
+  port: '443',
+  base: '/api/ipfs',
+  pinning: true,
+});
+
+const ipfs: any = {};
+
+ipfs.hashToHex = (hash: string): string => {
+  return '0x' + bs58.decode(hash).toString('hex').substring(4);
+}
+
+ipfs.hexToHash = (hex: string): string => {
+  if (hex.substring(0,2) === '0x') hex = hex.substring(2);
+  return bs58.encode(Buffer.from(ipfsHashPrefix + hex, 'hex'));
+}
+
+ipfs.add = (data: Object) => {
+  return promisify(ipfsApi.addJSON.bind(ipfsApi))(data);
+}
+
+const cache: any = {};
```

Comment:
Browsers cache API calls for us so caching like this isn't necessary client-side..
But this module also runs in node as part of the bg_cacher and in this environment it is worthwhile

Reply:
The bg_cacher will be phased out soon. 
Apart from browser cache, we are saving this info also in the redux state. So I'd remove the cachaing machinery here.